api/envoy/api/v2/lds.proto

```diff
@@ -203,4 +222,11 @@ message Listener {
   // <envoy_api_field_listener.UdpListenerConfig.udp_listener_name>` = "raw_udp_listener" for
   // creating a packet-oriented UDP listener. If not present, treat it as "raw_udp_listener".
   listener.UdpListenerConfig udp_listener_config = 18;
+
+  // [#not-implemented-hide:]
+  // Used only when the :ref:`type<envoy_api_field_Listener.type>` field is set to
+  // :ref:`API_HTTP<envoy_api_field_Listener.Type.API_HTTP>`.
+  // The HttpConnectionManager configuration to be used by the HTTP listener.
+  envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager
```

Comment:
WDYT about adding an `ApiListener` message, and then inside of that having a oneof with various types of API listeners, HCM being the only one today? I think @lizan already made this point, but in the future we might want to support TCP, Redis, etc.

Reply:
I've attempted to do this.  Please let me know if this is not what you had in mind.